Sources/ForageSDK/Foundation/Network/LiveForageService.swift

```diff
@@ -310,14 +310,23 @@ extension LiveForageService: Polling {
             completion(.failure(error))
         }
     }
+
+    /// We generate a random jitter amount to add to our retry delay when polling for the status of
+    /// Payments and Payment Methods so that we can avoid a thundering herd scenario in which there are
+    /// several requests retrying at the same exact time.
+    /// 
+    /// Returns a random double between -25 and 25  
+    internal func getJitterAmount() -> Double {
+        return Double(Int.random(in: -25...25)) / 1000.0
+    }
```

Comment:
I think we can use the 2nd `using` / `rng` param of `Int.random` to inject the mock random number generator in our unit test.

and default to the [SystemRandomNumberGenerator](https://developer.apple.com/documentation/swift/systemrandomnumbergenerator) when it's not provided (i.e. called in the LiveForageService.

maybe something like this

```swift
protocol RandomNumberGeneratorProtocol {
    mutating func next() -> UInt64
}

internal func getJitterAmount(randomNumberGenerator: RandomNumberGeneratorProtocol? = nil) -> Double {
    let rng = randomNumberGenerator ?? SystemRandomNumberGenerator() // fallback to system one
    return Double(Int.random(in: -25...25, using: &rng)) / 1000.0
}
```

In the tests, maybe something like this:

```swift
struct FixedSeedRandomNumberGenerator: RandomNumberGeneratorProtocol {
    mutating func next() -> UInt64 {
        return <insert-fixed-seed-here>
    }
}

// in test case:
getJitterAmount(randomNumberGenerator: FixedSeedRandomNumberGenerator())
```

Reply:
If this takes a while to figure out. I say we just add some print statements showcasing that this works as intended and merge this PR.

Reply:
Having some issues getting this to work, I'll go with my original approach for now, but we can discuss later further if I was perhaps missing something

Reply:
I'm not sure why the coverage is failing though, we can also discuss this when you get the chance